cadasta/mixin/network_mixin.py

```diff
@@ -0,0 +1,89 @@
+import logging
+import abc
+import json
+from qgis.PyQt.QtNetwork import (
+    QNetworkAccessManager,
+    QNetworkReply,
+    QNetworkRequest
+)
+from qgis.PyQt.QtCore import QUrl, QByteArray
+
+LOGGER = logging.getLogger('CadastaQGISPlugin')
+
+__author__ = 'dimas.ciputra@gmail.com'
+__date__ = '15/12/2016'
+
+
+class NetworkMixin(object):
+    """A mixin that can be used to send network request and receive replies."""
+
+    request_url = str
+    error = None
+    results = None
+
+    def __init__(self, request_url=None):
+        if request_url is not None:
+            self.request_url = request_url
+        self.manager = QNetworkAccessManager()
```

Comment:
Yes @dimasciput, thanks @Gustry. This is critical - its the first thing I asked @cchristelis to file in your issue tracker: #1